pcs/src/traits.rs

```diff
@@ -0,0 +1,63 @@
+use arith::{Field, FieldSerde};
+use rand::RngCore;
+use std::fmt::Debug;
+
+pub trait PCS<F: Field + FieldSerde> {
+    type Params: Clone + Debug;
+    type Poly: Clone + Debug;
+    type EvalPoint: Clone + Debug;
+
+    type SRS: Clone + Debug + FieldSerde;
+    type PKey: Clone + Debug + From<Self::SRS> + FieldSerde;
+    type VKey: Clone + Debug + From<Self::SRS> + FieldSerde;
+    type Commitment: Clone + Debug + FieldSerde;
+    type Opening: Clone + Debug + FieldSerde;
+
+    fn gen_srs_for_testing(&self, rng: impl RngCore, params: &Self::Params) -> Self::SRS;
+
+    fn commit(
+        &self,
+        params: &Self::Params,
+        proving_key: &Self::PKey,
+        poly: &Self::Poly,
+    ) -> Self::Commitment;
```

Comment:
I actually have a minor thought here: can we separate `Commitment` into `CommitmentWithData` and `Commitment`?

The thinking process proceeds as follows: supposing you are doing a FRI-based pcs or a linear-code tensor-IOP based pcs, then the commitment itself should be just a MT hash, while commitment with data contains all the codewords related.  In this case, the `commit` method should output a `CommitmentWithData` containing all the MTs committing against related codewords.

Take a step back, say we are using a KZG-alike curve based commitment, then `CommitmentWithData` should be equivalent to `Commitment`, as it falls back to the crs.

(Simply just dumping my thoughts here, but I don't expect you do take further action on this - maybe we should just merge in this one and I will do some modifications on top of it.)